.github/workflows/coverage_runner.yml

```diff
@@ -0,0 +1,109 @@
+        description: Enter PR number coming from community to run coverage on it. Do not run for PR coming from hazelcast organization, they will be run automatically.
+        required: true
+  push:
+    branches:
+      - master
+      - '[45].*.z'
+  pull_request_target:
+    branches:
+      - master
+      - '[45].*.z'
+    types:
+      [opened]
+jobs:
+  check_for_membership:
+    runs-on: ubuntu-latest
+    name: Check membership of given user  
+    outputs:
+      check-result: ${{ steps.composite.outputs.check-result }}
+    steps: 
+      - name: Action for membership check
+        id: composite
+        uses: hazelcast/hazelcast-tpm/membership@main
+        with:
+          organization-name: 'hazelcast'
+          member-name: ${{ github.event.pull_request.head.repo.owner.login }}
+          token: ${{ secrets.PAT }}
+          
+  run-tests:
+    runs-on: ${{ matrix.os }}
+    needs: [check_for_membership]
+    if: github.event_name == 'push' || needs.check_for_membership.outputs.check-result == 'true' || github.event_name == 'workflow_dispatch'
+    name: Run tests with Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.10' ]
+        os: [ ubuntu-latest, windows-latest ]
+      fail-fast: false
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+          
+      - name: Checkout code for push event
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+
+      - name: Checkout code for PR
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          
+      - name: Checkout repository to fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v2
+        
+      - name: Fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git fetch origin pull/${{github.event.inputs.pr_number }}/head:temp_branch
+          git checkout temp_branch
+          
+      - name: Install dependencies
+        run: |
+          pip install -r requirements-test.txt
+      - name: Run tests
+        env:
+          HAZELCAST_ENTERPRISE_KEY: ${{ secrets.HAZELCAST_ENTERPRISE_KEY }}
+        run: python run_tests.py
+            
+      - name: Publish results to Codecov for PR coming from hazelcast organization
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' &&  github.event_name == 'pull_request_target' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          override_pr: ${{ github.event.pull_request.number }}
```

Comment:
Maybe codecov action can determine the PR number itself. In the previous version of the file we were not specifying the PR number in `pull_request` event. They may have that support for `pull_request_target` as well.  If so we can merge two of these steps. This one and the one below.

Reply:
Most probably when we remove `override_pr: ${{ github.event.pull_request.number }}` this line it will work but as the event trigger name is different, I feel more safe to specify PR number to the codecov action also. But If you are not happy with it, I can test it and if it works we can remove it. But I prefer to keep it.

Reply:
Ok we can keep it, I was just curious.